specification/storage/resource-manager/Microsoft.Storage/stable/2021-09-01/common.json

```diff
@@ -84,6 +84,10 @@
           "minimum": 1,
           "maximum": 365,
           "description": "Indicates the number of days that the deleted item should be retained. The minimum specified value can be 1 and the maximum value can be 365."
+        },
+        "allowPermanentDelete": {
+          "type": "boolean",
+          "description": "This property when set to true allows deletion of the soft deleted blob versions and snapshots. This property cannot be used blob restore policy."
```

Comment:
This DeleteRetentionPolicy will be referenced by Blob , container, fileshare. Are all 3 object support DeleteRetentionPolicy ?
If so, we should add examples for all 3. 
If not, we should limit to change to supported object.

Reply:
It is supported for only Blob today and not others,   We may support it in the future with the same model. The Server will not accept non-blob service properties.

Reply:
If "The Server will not accept non-blob service properties.", would you please don't add it for container/fileshare, to avoid misusage from customer, which might take customer issue.

Reply:
I understand, I can update the description over here, it's a breaking change to create a new Class for blob objects ? We do the same for LCM as well, we have common models but we provide info via description and server-side handles error messages.
Are you okay with this?

Reply:
I am OK if the description can clearly describe the new property only work for blob.
You should also get ARM review board agree on this.

Reply:
Addressed